src/pip/_internal/index.py

```diff
@@ -288,22 +317,10 @@ def iter_all(self):
 
     def iter_applicable(self):
         # type: () -> Iterable[InstallationCandidate]
-        """Iterate through candidates matching the given specifier.
+        """Iterate through candidates matching the given versions.
```

Comment:
I would say something other than "given" here because the versions are associated with the object rather than given as arguments (which is what given normally means in the context of a function or method docstring).

Reply:
Is the current wording better? (I am not good at describing things.)

Reply:
It's not a big deal. But some other possibilities that come to mind: "Iterate through the candidates matching the instance's versions." or "matching the versions associated with the instance" or "matching the associated versions"